pkg/networkservice/core/trace/client.go

```diff
@@ -27,21 +27,35 @@ import (
 
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 	"github.com/networkservicemesh/sdk/pkg/tools/logger"
-
 	"github.com/networkservicemesh/sdk/pkg/tools/typeutils"
 )
 
 type traceClient struct {
 	traced networkservice.NetworkServiceClient
 }
 
+type printerClient struct{}
+
+func (t *printerClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	logRequest(ctx, request)
+	conn, err := next.Client(ctx).Request(ctx, request, opts...)
+	logResponse(ctx, conn)
+	return conn, err
+}
+
+func (t *printerClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
+	logRequest(ctx, conn)
+	return next.Client(ctx).Close(ctx, conn, opts...)
```

Comment:
Why we do not log response here?

Reply:
I think because `Close(...)` returns `protobuf.Empty`. There is nothing for logging.

Reply:
Yes, but can we try to log changes in conn?

Reply:
Probably you are right.
Do you think there might be the next implementation of Close?
```
func Close(...) {
	// some code
	r, err := next.Server(ctx).Close(ctx, connection)
	// changes inside connection here
	return r, err
}
```